### PR TITLE
W-15609157: Workaround to avoid harcoding the product version in Mule…

### PR DESCRIPTION
**Pull request by anosenzo**
…Manifest.java#getProductVersion()

### CODE REVIEW

File: core/pom.xml [hunk 1]
```diff
@@ -19,19 +19,27 @@
...
+        <resources>
+            <resource>
+                <directory>src/main/resources</directory>
+                <includes>
+                    <include>**/version.properties</include>
+                </includes>
+                <filtering>true</filtering>
+            </resource>
+            <resource>
+                <directory>src/main/resources</directory>
```
Comment (elrodro83) [inline core/pom.xml:36]: this is the same as the one above

File: core/pom.xml [hunk 2]
```diff
@@ -19,19 +19,27 @@
...
-                <artifactId>maven-site-plugin</artifactId>
-                <configuration>
-                    <skip>true</skip>
-                    <skipDeploy>true</skipDeploy>
-                </configuration>
-            </plugin>
-        </plugins>
+        <resources>
+            <resource>
+                <directory>src/main/resources</directory>
```
Comment (elrodro83) [inline core/pom.xml:29]: use a different folder for filtering resources, to prevent accidental repmacements on the base resources

**Review by elrodro83 — CHANGES_REQUESTED** [review]

File: core/pom.xml (hunk 2 above)
Comment (anosenzo) [reply core/pom.xml:29]: Rodro I have try some tests and the directory tag is a required one. It sets where the resources folders is going to be present so we should maintain the default "src/main/resources". 

In the include I have moved the properties file to a new folder product-version.

File: core/pom.xml (hunk 1 above)
Comment (anosenzo) [reply core/pom.xml:36]: Same question as https://github.com/mulesoft/mule/pull/13644/files#r1685004540

File: core/pom.xml (hunk 2 above)
Comment (elrodro83) [reply core/pom.xml:29]: why not create a `src/main/resources-filtering` for this?

**Review by elrodro83 — CHANGES_REQUESTED** [review]

File: core/pom.xml (hunk 2 above)
Comment (anosenzo) [reply core/pom.xml:29]: I have done what you suggested, maintain the original src/main/resources without filtering and add a new src/main/resources-filtering with the filter in true and it worked fine.

**Review by elrodro83 — APPROVED** [review]

**Pull request merged**